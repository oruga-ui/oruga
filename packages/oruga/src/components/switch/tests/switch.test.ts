import { describe, test, expect, afterEach } from "vitest";
import { enableAutoUnmount, mount } from "@vue/test-utils";

import OSwitch from "@/components/switch/Switch.vue";

describe("OSwitch tests", () => {
    enableAutoUnmount(afterEach);

    test("render correctly", () => {
        const wrapper = mount(OSwitch);
        expect(!!wrapper.vm).toBeTruthy();
        expect(wrapper.exists()).toBeTruthy();
        expect(wrapper.attributes("data-oruga")).toBe("switch");
        expect(wrapper.html()).toMatchSnapshot();
    });
});
