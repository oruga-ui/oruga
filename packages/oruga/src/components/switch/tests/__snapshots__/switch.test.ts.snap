// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`OSwitch tests > render correctly 1`] = `
"<label class="o-switch o-switch--right" data-oruga="switch" role="switch" aria-checked="false"><input type="checkbox" role="switch" data-oruga-input="switch" class="o-switch__input" autocomplete="off" true-value="true" false-value="false"><span class="o-switch__check o-switch--rounded"><span class="o-switch__check-switch o-switch--rounded"></span></span>
  <!--v-if-->
</label>"
`;
